poller/poller.go

```diff
@@ -0,0 +1,58 @@
+package poller
+
+import (
+	"time"
+
+	"github.com/godaddy/split-go-serializer/api"
+)
+
+// Poller contains cache data and SplitioAPIBinding
+type Poller struct {
+	Cache              string
+	SplitioAPIBinding  api.SplitioAPIBinding
+	PollingRateSeconds int
+	SerializeSegments  bool
+	quit               chan bool
+}
+
+// NewPoller returns a new Poller
+func NewPoller(splitioAPIKey string, pollingRateSeconds int, serializeSegments bool) *Poller {
+	if pollingRateSeconds == 0 {
+		pollingRateSeconds = 300
+	}
+	splitioAPIBinding := api.NewSplitioAPIBinding(splitioAPIKey, "")
+	return &Poller{"", *splitioAPIBinding, pollingRateSeconds, serializeSegments, make(chan bool)}
+}
+
+// pollForChanges will get the latest data of splits and segment
+func (poller *Poller) pollForChanges() error {
+	//TODO: call getSplits and getSegments to formulate the cache
+	poller.Cache = "data from splitChanges and segmentChanges"
+	return nil
+}
+
+// Poll creates a goroutine and keep tracking until it stops
+func (poller *Poller) Poll() {
+	go poller.jobs()
+	<-poller.quit
```

Comment:
why read from the channel?

Reply:
I thought otherwise the original main routine won't wait for the go routine in jobs() finished and the `Cache` won't be updated

Reply:
I think I understand: you want to implement a mechanism that signals when _some_ splits + segments have been loaded into the local cache?

Reply:
I think I want `pollForChanges()` to be called continuously until `Stop()` is called and the `quit` is set to true. If not read the quit here, I think this function will end, not waiting for jobs() completed.

Reply:
why do you need to wait for `jobs` to complete?

Reply:
oh I am considering when the client call `Poll()` for the first time and directly call `getSerializerData()`,  the cache may not be updated yet (means it is empty) because Poll won't wait for this new go routine.  
Probably I just need to make Poll() sleep for like 2 seconds? (to avoid having empty data for the first time)

Reply:
yea, I don't think we want to wait for `jobs` to complete. `jobs` will run in an infinite loop until `Stop` is called

Reply:
if you want to want code to block in `getSerializerData()` (or something else like `waitUntilReady()`) use a condition variable.  some psuedo code:

```go
m sync.Mutex
c := sync.NewCond(&m)
...

if (!cacheHasContents) {
  m.Lock()
  if (!cacheHasContents) {
    c.Wait()
  }
  m.Unlock()
}
```

```go
cacheContents = {}
cacheHasContents = true
c.Broadcast()
```